src/github.ts

```diff
@@ -1788,9 +1851,200 @@ export class GitHub {
+    }
+
+    if (currentLockRule.repository.ref.branchProtectionRule.lockBranch) {
+      this.logger.warn(`Branch '${branchName}' was already locked`);
+      return;
+    }
+
+    this.logger.info(`Locking branch '${branchName}', it is now read-only`);
+    await this.mutateLockBranchProtectionRule(
+      currentLockRule.repository.ref.branchProtectionRule.id,
+      true
+    );
+  }
+
+  async unlockBranch(branchName: string) {
+    const currentLockRule = await this.queryLockBranchProtectionRule(
+      branchName
+    );
+    if (!currentLockRule) {
+      this.logger.warn(
+        `No lock protection rule found for branch '${branchName}'`
+      );
+      return;
+    }
+    if (!currentLockRule.repository.ref.branchProtectionRule.lockBranch) {
+      this.logger.warn(`Branch '${branchName}' was already unlocked`);
+      return;
+    }
+
+    this.logger.info(`Unlocking branch '${branchName}', it now allows writes`);
+    await this.mutateLockBranchProtectionRule(
+      currentLockRule.repository.ref.branchProtectionRule.id,
+      false
+    );
+  }
+
+  private async queryLockBranchProtectionRule(
+    branchName: string
+  ): Promise<GraphQLLockBranchProtectionRule | null> {
+    const query = `query lockBranchProtectionRule($owner: String!, $repo: String!, $branchName: String!) {
+        repository(name: $repo, owner: $owner) {
+          ref(qualifiedName: $branchName) {
+            branchProtectionRule {
+              id
+              lockBranch
+            }
+          }
+        }
+      }`;
+    const currentProtectionRule = await this.graphqlRequest({
+      query,
+      owner: this.repository.owner,
+      repo: this.repository.repo,
+      branchName,
+    });
+
+    if (!currentProtectionRule?.repository?.ref?.branchProtectionRule?.id) {
+      return null;
+    }
+    return currentProtectionRule as GraphQLLockBranchProtectionRule;
+  }
+
+  private async mutateLockBranchProtectionRule(
+    protectionRuleId: string,
+    locked: boolean
+  ) {
+    const mutation = `mutation MutateLockBranch($ruleId: ID!, $locked: Boolean) {
+        updateBranchProtectionRule(
+          input: {branchProtectionRuleId: $ruleId, lockBranch: $locked}
+        ) {
+          branchProtectionRule {
+            lockBranch
+          }
+        }
+      }`;
+    await this.graphqlRequest({
+      query: mutation,
+      ruleId: protectionRuleId,
+      locked: locked,
+    });
+  }
+
+  private async createLockBranchProtectionRule(
+    branchName: string,
+    locked: boolean
+  ): Promise<GraphQLLockBranchProtectionRule | null> {
+    const repositoryResponse = await this.octokit.repos.get({
+      owner: this.repository.owner,
+      repo: this.repository.repo,
+    });
+
+    const mutation = `mutation CreateLockBranch($repositoryId: ID!, $branchName: String!, $locked: Boolean) {
+        createBranchProtectionRule(
+          input: {repositoryId: $repositoryId, pattern: $branchName, lockBranch: $locked, allowsForcePushes: true}
```

Comment:
Note the input `allowsForcePushes: true`. This setting defaults to false when a branch protection rule is created, but we need to be able to force push to re-align the branch `next` with `main`.